src/core/thread/link_quality.hpp

```diff
@@ -94,23 +94,23 @@ class SuccessRateTracker
      * @retval the average failure rate `[0-kMaxRateValue]` with `kMaxRateValue` corresponding to 100%.
      *
      */
-    uint16_t GetFailureRate(void) const { return kMaxRateValue - mSuccessRate; }
+    uint16_t GetFailureRate(void) const { return mFailureRate; }
 
     /**
      * This method returns the average success rate.
      *
      * @retval the average success rate as [0-kMaxRateValue] with `kMaxRateValue` corresponding to 100%.
      *
      */
-    uint16_t GetSuccessRate(void) const { return mSuccessRate; }
+    uint16_t GetSuccessRate(void) const { return kMaxRateValue - mFailureRate; }
 
 private:
     enum
     {
         kDefaultWeight = 64,
     };
 
-    uint16_t mSuccessRate;
+    uint16_t mFailureRate;
```

Comment:
Changed the `SuccessRateTracker` to store `mFailureRate` so that it can be cleared when it's `memset()` to zero. Noe that `SuccessRateTracker` is now used in `Neighbor`, `Child` and `Router` structs , and we use `memset(0)` to clear the entire child/router entry.